### PR TITLE
Close Issues from Archived Repositories

### LINKED ISSUE #32: Close issues from archived repositories
There may be cases when issue exists in an archived repository (i.e. read only) so issues can't be self assigned there although all issue labels would be totally valid.

What should be done:
- if project issue exists in an archived partner's repository then the bot should close such issue

### PR DESCRIPTION
**Pull request by jordan-ae**
Resolves [#32](https://github.com/ubiquity/devpool-directory-bounties/issues/32)

### CODE REVIEW

File: helpers/github.ts
Comment (0x4007) [inline helpers/github.ts]: projectIssue.repository?

dont use ?

catch and throw an error

Comment (jordan-ae) [reply helpers/github.ts]: Do you mean something like this?
```
  if (!projectIssue.repository || projectIssue.repository.archived === true) {
    if (!projectIssue.repository) {
      throw new Error("Repository data is missing for the project issue.");
    }
    return;
  }
```

**Comment by jordan-ae** [comment]
@0x4007 is this okay?

**Comment by jordan-ae** [comment]
@0x4007 can I please get a review here?

File: helpers/github.ts
Comment (0x4007) [inline helpers/github.ts:382]: Checking the same condition twice doesn't make sense you need to refactor your logic

File: tests/devpool-issue-handler.test.ts
Comment (0x4007) [inline tests/devpool-issue-handler.test.ts:1447]: All of your changes only check for archived false. Is that right?

**Review by 0x4007 — CHANGES_REQUESTED** [review]
You need to post a QA test before approval.

**Pull request closed without merging**
